src/app/clusters/ota-requestor/GenericOTARequestorDriver.cpp

```diff
@@ -146,7 +147,12 @@ void GenericOTARequestorDriver::UpdateNotFound(UpdateNotFoundReason reason, Syst
         break;
     case UpdateNotFoundReason::kBusy:
         willTryAnotherQuery = true;
-        break;
+        if (mProviderRetryCount++ <= kMaxBusyProviderRetryCount)
+        {
+            break;
+        }
+        ChipLogProgress(SoftwareUpdate, "Max Busy Provider retries reached. Attempting to get next Provider.");
+        [[fallthrough]]; // Else, fall through and select another provider
```

Comment:
Looks like this is a C++17 feature, last I heard Matter is at C++14

Reply:
It appears the other options are to use [[gnu::fallthrough]]; (C++11 and 14) or comment markers. Will add them.